R/kernel.r

```diff
@@ -166,6 +166,27 @@ abort_queued_messages = function() {
 
 },
 
+handle_stdin = function() {
+    "React to a stdin message coming in"
+
+    # wait for 'input_reply' response message
+    while (TRUE) {
+        log_debug('stdin loop: beginning')
+        zmq.poll(c(sockets$stdin),          # only stdin channel
+                 c(.pbd_env$ZMQ.PO$POLLIN)) # type
+
+        if (bitwAnd(zmq.poll.get.revents(1), .pbd_env$ZMQ.PO$POLLIN)) {
+            log_debug('stdin loop: found msg')
+            parts <- zmq.recv.multipart(sockets$stdin, unserialize = FALSE)
+            msg <- wire_to_msg(parts)
+            return(msg$content$value)
+        } else {
+            # else shouldn't be possible
+            log_debug('stdin loop: zmq.poll returned but no message found?')
```

Comment:
this should be `log_error`. i’d also prefer if you do the following instead of “early” exit by `return`

```r
if (...) {
    ...
    msg$content$value
} else {
    # ...
    log_error(...)
    NULL
}